addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -90,6 +82,24 @@ export class LinkTools extends Link {
             this.props.onDestroy();
         });
     }
+    /**
+     * @override
+     */
+    async willUpdateProps(newProps) {
+        await super.willUpdateProps(newProps);
+        await this.mountedPromise;
+        this.$link = newProps.link ? $(newProps.link) : this.link;
+        this._setSelectOptionFromLink();
+        this._updateOptionsUI();
+        this._updateLabelInput();
+
+        if (!this.linkEl.href && this.state.url) {
+            // Link URL was deduced from label. Apply changes to DOM.
+            this.__onURLInput();
+        }
```

Comment:
Do you have an example where this is useful? I had the feeling that the one in the `start()` was enough.